management-app/src/pages/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,63 @@
+import { Alert, CircularProgress, Grid } from '@mui/material';
+import { useCallback, useEffect } from 'react';
+
+import AddBoardBtn from './components/AddBoardBtn/AddBoardBtn';
+import Board from './components/Board/Board';
+
+import BoardModal from './components/BoardModal/BoardModal';
+import { useAppDispatch, useAppSelector } from '../../store';
+import { closeBoardModal, getBoards, openBoardModal } from './slice/mainSlice';
+
+function MainPage() {
+  const boards = useAppSelector((state) => state.main.boards);
+  const isBoardModalOpen = useAppSelector(
+    (state) => state.main.isBoardModalOpen
+  );
```

Comment:
I think isBoardModalOpen should be in the local MainPage state. Same applies to isLoading and isError.